### DISCUSSION TITLE
[Feature Request] SYNC Consume/Export folders with a local folder

### DISCUSSION THREAD

**Discussion by thiscantbeserious** [astubenbord/paperless-mobile#198] [Feature Requests]
Story:

It would be very useful to have a sync feature that would allow you to access files while the Paperless-Server is offline/unreachable.

For example when working from remote, or when your Paperless Server is only reachable via VPN. Or when you're in an untrusted Subnet (other VPN, for example).

That would allow you to sync both the consume and export folder with a folder on your Mobile Phone.

Technical:

This should not require any kind of local DB on your device for offline access, since you're operating on the file level - just reusing the consume/export folder concept.

In addition to that, you could add two kinds of filters: Include and Exclude. By Label, by Document-Type and by Filename (Pattern)

Possible troubles:

When using multiple Paperless Servers. Not sure if thats a thing, or if you have already solved that via configuration splitting of some sort.

**Comment by astubenbord** [comment]
Hi, thanks for creating a feature request :)

I agree that having documents available offline is a great idea and can be very useful. The way I initially planned to support this is to let a user select documents which can then be downloaded to the device and a reference to the storage path would be stored in an on-device database (e.g. in hive). 

With multiple users, we have to either always delete the documents made available offline on an account switch or store them encrypted in namespaces, such that if someone found your phone they can't just read the documents. Also, even without multiple users, you probably don't want your files to just be stored on your phone, especially with biometric authentication enabled. Therefore I think the local db approach would be easier and more secure, as the app already uses a local database for persistence of settings, user accounts, credentials and app state in general.

But I'd like to open a disucssion once I have time to implement it to discuss this in more detail, and this is one of the features with higher priority like the upcoming permission update and the custom document scanner. 

Files can also be out of date or be deleted on the server which cannot directly be registered in the app (maybe there's a task list which can be queried for deletion events or so, I don't know), so we'd also require some sort of syncing mechanism, and that's where things probably get more complicated.


Cheers

**Reply by beyercenter** [reply]
Why not use another App for this? Syncthing is great for this.

But of courseyou need to export something beforeyou can sync those files...

Your consumefolder should be empty?!

**Reply by MrModest** [reply]
@beyercenter  I'd argue that having just a list of docs isn't enough. I'd love to see full offline support of my documents, like it's implemented in 1Password, for example. I have full access to my credentials when offline; I only can't add/edit them, which is fine.

**Comment by MrModest** [comment]
Hi there! Are there any updates on this topic? 

It would be great to have an option to have an unencrypted cache (at the user's own risk) with a warning message of so.. 

Because, I assume, implementing fully encrypted features could take a lot of time

**Comment by astubenbord** [comment]
Please continue the discussion in #102

**Reply by MrModest** [reply]
That issue is about custom fields, but this one is about offline access. How are they related? 🤔

**Reply by astubenbord** [reply]
Ha, sorry, I meant #102. I corrected it.
